src/core/ext/filters/client_channel/lb_policy/round_robin/round_robin.c

```diff
@@ -761,12 +759,13 @@ static void rr_update_locked(grpc_exec_ctx *exec_ctx, grpc_lb_policy *policy,
     return;
   }
   grpc_lb_addresses *addresses = arg->value.pointer.p;
-  size_t num_addrs = 0;
   for (size_t i = 0; i < addresses->num_addresses; i++) {
-    if (!addresses->addresses[i].is_balancer) ++num_addrs;
+    // If there were any balancer, we would have chosen grpclb policy instead.
+    GPR_ASSERT(!addresses->addresses[i].is_balancer);
```

Comment:
Same comment here as what David suggested in the PF code: this assertion can be moved to line 800, and this for loop can go away completely.